packages/nuxt/test/devonly.test.ts

```diff
@@ -0,0 +1,41 @@
+import { describe, expect, it } from 'vitest'
+import type { Plugin } from 'vite'
+import { DevOnlyPlugin } from '../src/core/plugins/dev-only'
+import { normalizeLineEndings } from './utils'
+const pluginVite = DevOnlyPlugin.raw({}, { framework: 'vite' }) as Plugin
+
+const viteTransform = async (source: string, id: string) => {
+  const result = await (pluginVite.transform! as Function)(source, id)
+  return typeof result === 'string' ? result : result?.code
+}
+
+describe('test devonly transform ', () => {
+  it('test dev only treeshaking', async () => {
+    const result = await viteTransform(`<template>
+    <div>
+      <DevOnly>
+        <SomeDevOnlyComponent></SomeDevOnlyComponent>   
+      </DevOnly>
+    </div>
+    <SomeComponent>
+        <dev-only>
+          test
+        </dev-only>
+    </SomeComponent>
+    </template>`, 'some id')
+
+    expect(normalizeLineEndings(result)).toMatchInlineSnapshot(`
+      "<template>
+          <div>
+            
+          </div>
+          <SomeComponent>
+              
+          </SomeComponent>
+          </template>"
+    `)
+
+    expect(result).not.toContain('dev-only')
+    expect(result).not.toContain('DevOnly')
+  })
```

Comment:
```suggestion
   expect(result).not.toContain('lazy-dev-only')
   expect(result).not.toContain('LazyDevOnly')
 })
```
This makes sure lazy variation isn't found either.